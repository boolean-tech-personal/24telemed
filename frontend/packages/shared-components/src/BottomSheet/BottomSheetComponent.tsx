import styled, { css } from 'styled-components';
import { BottomSheetCancelIcon} from '../Icon';
import { Typography } from '../Typography';


export interface BottomSheetComponentProps {
  children?: React.ReactNode;
  title?: string;
  onClickCancel: () => void;
}

export function BottomSheetComponent({
  onClickCancel,
  children,
  title
}: BottomSheetComponentProps) {
  return (
    <Container>
      <LabelAndCancelContainer>
        <Content>
          {title && <Title>
            <Typography variant="bodyMd">{title}</Typography>
            
            </Title>}
          <CancelIconContainer onClick={onClickCancel}>
            <BottomSheetCancelIcon />
          </CancelIconContainer>
          {children}
        </Content>
      </LabelAndCancelContainer>
    </Container>
  );
}

const Container = styled.div`
  position: relative;
  width: 100%;
  box-shadow: ${({ theme }) => theme.shadows.md};
  padding: ${({ theme }) => theme.spacing.xl};
  box-sizing: border-box;

  border-radius: ${({ theme }) => theme.radius.card} ${({ theme }) => theme.radius.card} 0 0;
  border: ${({ theme }) => theme.border.primary.light};
  background-color: ${({ theme }) => theme.palette.common.white};
  
`;

const Content = styled.div`
  margin-bottom: ${({ theme }) => theme.spacing.lg};
  margin-top: ${({ theme }) => theme.spacing.lg};
`;

const LabelAndCancelContainer = styled.div`
  display: flex;
  justify-content: space-between;
`;

const Title  = styled.div`
  margin-bottom: ${({ theme }) => theme.spacing.sm};
  position: absolute;
  top: ${({ theme }) => theme.spacing.sm};
  font-weight: ${({ theme }) => theme.typography.fontWeight.bold};  
  
`;

const CancelIconContainer = styled.div`
  position: absolute;
  right: ${({ theme }) => theme.spacing.sm};
  top: ${({ theme }) => theme.spacing.sm};
  cursor: pointer;
  :hover {
    transform: scale(1.1);
  }
  
`;
