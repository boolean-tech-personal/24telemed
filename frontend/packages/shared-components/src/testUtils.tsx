import React, { ReactElement } from "react";

import "@testing-library/jest-dom/vitest";
import { RenderOptions, render } from "@testing-library/react";
import "jest-styled-components";
import { ThemeProvider } from "styled-components";
import { vi } from "vitest";

import { defaultTheme } from "./styles/defaultTheme";

export * from "./testHelpers";

const AllTheProviders = ({ children }: { children: ReactElement }) => {
  return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
  // @ts-ignore
) => render(ui, { wrapper: AllTheProviders, ...options });

const matchMedia = (mediaRule: string) =>
  vi.fn((media: string) => ({
    matches: media === mediaRule,
    media,
    addEventListener: vi.fn(),
    removeEventListener: vi.fn(),
  }));

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, matchMedia };
