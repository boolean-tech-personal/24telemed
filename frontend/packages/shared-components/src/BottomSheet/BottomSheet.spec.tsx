import { render, screen } from "../testUtils";
import {BottomSheetComponent } from './BottomSheetComponent'

describe("BottomSheet", () => {
  it("should render properly", () => {
    
    render(
      <BottomSheetComponent onClickCancel={() => {}}>
        <div>BottomSheet Content</div>
      </BottomSheetComponent>
    );
    const el = screen.getByText("BottomSheet Content");
    expect(el).toBeInTheDocument();
    expect(el).toMatchSnapshot();
  });

  it("should render title properly", () => {
    render(
      <BottomSheetComponent onClickCancel={() => {}} title="BottomSheet Title">
        <div>BottomSheet Content</div>
      </BottomSheetComponent>
    );
    const el = screen.getByText("BottomSheet Title");
    expect(el).toBeInTheDocument();
    expect(el).toMatchSnapshot();
  });

});
