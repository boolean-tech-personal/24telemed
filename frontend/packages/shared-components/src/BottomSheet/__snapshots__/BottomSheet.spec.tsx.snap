// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BottomSheet > matches snapshot 1`] = `
<div>
  <div
    class="sc-gswNZR bhtNKK"
  >
    <div
      class="sc-hLBbgP kBCcYD"
    >
      <div
        class="sc-dkrFOg btMqjT"
      >
        <div
          class="sc-eDvSVe bXPsfw"
        >
          Filter
        </div>
        <div
          class="sc-jSUZER iFgFoH"
        >
          <i
            class="sc-bcXHqe bgEkPo"
            data-testid="BottomSheetCancelIcon"
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_4862_2303)"
              >
                <path
                  d="M18 6L6 18"
                  stroke="black"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.75"
                />
                <path
                  d="M6 6L18 18"
                  stroke="black"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.75"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_4862_2303"
                >
                  <rect
                    fill="white"
                    height="24"
                    width="24"
                  />
                </clippath>
              </defs>
            </svg>
          </i>
        </div>
        This is a bottom sheet content
      </div>
    </div>
  </div>
</div>
`;

exports[`BottomSheet > should render properly 1`] = `
<div>
  BottomSheet Content
</div>
`;

exports[`BottomSheet > should render title properly 1`] = `
.c0 {
  color: #2F4D52;
  text-align: left;
  width: auto;
  margin-top: 0;
  margin-bottom: 0px;
  font-family: "Uniform Rounded","Helvetica","Arial",sans-serif;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  overflow-wrap: break-word;
}

.c0 > a {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 700;
}

<p
  class="c0"
>
  BottomSheet Title
</p>
`;
